flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -1195,6 +1197,55 @@ public void testSlotSharingOnAllVerticesInSameSlotSharingGroupByDefaultDisabled(
         assertDistinctSharingGroups(source1Vertex, source2Vertex, map2Vertex);
     }
 
+    @Test
+    public void testSlotSharingResourceConfiguration() {
+        final String slotSharingGroup1 = "slot-a";
+        final String slotSharingGroup2 = "slot-b";
+        final ResourceProfile resourceProfile1 = ResourceProfile.fromResources(1, 10);
+        final ResourceProfile resourceProfile2 = ResourceProfile.fromResources(2, 20);
+        final ResourceProfile resourceProfile3 = ResourceProfile.fromResources(3, 30);
+        final Map<String, ResourceProfile> slotSharingGroupResource = new HashMap<>();
+        slotSharingGroupResource.put(slotSharingGroup1, resourceProfile1);
+        slotSharingGroupResource.put(slotSharingGroup2, resourceProfile2);
+        slotSharingGroupResource.put(
+                StreamGraphGenerator.DEFAULT_SLOT_SHARING_GROUP, resourceProfile3);
+
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.fromElements(1, 2, 3)
+                .name(slotSharingGroup1)
+                .slotSharingGroup(slotSharingGroup1)
+                .map(x -> x + 1)
+                .name(slotSharingGroup2)
+                .slotSharingGroup(slotSharingGroup2)
+                .map(x -> x * x)
+                .name(StreamGraphGenerator.DEFAULT_SLOT_SHARING_GROUP)
+                .slotSharingGroup(StreamGraphGenerator.DEFAULT_SLOT_SHARING_GROUP);
```

Comment:
I think we need another test case for an operator that does not specify SSG (thus the default group) while the resource profile of the default group is specified.